src/terms/local.jl

```diff
@@ -36,7 +36,9 @@ struct ExternalFromReal
 end
 
 function (external::ExternalFromReal)(basis::PlaneWaveBasis{T}) where {T}
-    pot_real = external.potential.(r_vectors_cart(basis))
+    # Move this operation on CPU as GPU broadcast is very strict on type stability
+    rv = to_cpu(r_vectors_cart(basis))
+    pot_real = to_device(basis.architecture, external.potential.(rv))
```

Comment:
I don't think this is a good idea. It's better to keep thinks on the GPU and force users to code up their `external.potential` function properly. Otherwise you are making it impossible to do this sort of evaluation *only* on the GPU, which IMO is a very bad design decision.

Reply:
Thanks. I have removed it.